### DISCUSSION TITLE
MicroPython for XIAO ESP32S3 Sense | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#638] [Q&A]
# MicroPython for XIAO ESP32S3 Sense | Seeed Studio Wiki

MicroPython for XIAO ESP32S3 Sense

https://wiki.seeedstudio.com/XIAO_ESP32S3_Micropython/

**Comment by varmas18** [comment]


**Reply by CocoaBob** [reply]
I just had the same Wi-Fi issue, until I connected the antenna which came together in the package 😂

**Comment by starmice** [comment]
Where can i get the camera library? The sense firmware is only a binary with examples. I want to use my own micropython firmware and not be dependent on someone else to build the firmware for me.

**Comment by pierdolonez** [comment]
Hello,

I've been conducting tests related to MicroPython and camera handling, and I've noticed that at lower resolutions, the lowest values I can achieve for capturing an image to a variable are around 0.2 - 0.3 seconds.

For SVGA resolution, the camera documentation states that it should generate 30fps, which is currently unattainable in my tests. The maximum achievable frame rate is not even 3 fps.

When saving frames to JPG, this time is even worse. Is this an issue with MicroPython as an environment, or is the camera handling not optimized? I haven't conducted tests in the Arduino IDE, but I assume it might perform better there. Nevertheless, programming in Python/MicroPython is more convenient for me.

**Comment by nater3796** [comment]
I bought this assuming the basic documentation would at least get it going. That was not the case with this brand. The instructions were designed for people like the kid in the pic, who already knows how to do it, not for dummy's like me. It's a shame because I wanted to buy a ton of stuff to build security stuff for all my buildings but this brand couldn't put the same level of consideration toward training, as it does their awesome array of sensors. Not only that but I finally learned you have to restart it multiple times. Even if you just restarted it, it will flip back and fourth between not being able to connect to the cam and not being able to connect to the wifi. I would ask Seed Studio not to worry about replying to this thread but I see that's also something they don't do! Cheers!

**Reply by feiticeir0** [reply]
Hello ! I'm not affiliated with Seeed Studio or whatsoever, but what are you talking about ? 
You came here, you just write a bunch of stuff and you don't say what's wrong. 
What's your problem ? What are you trying to do that you  aren't accomplishing ? 
Seeed Studio has one of the most comprehensive wikis out there. A lot of examples, a lot of code and explaining. 

Just write what are you trying to do, what errors (if any) do you encounter and maybe someone will help you - because it might  have happened to it as well.
